### PR TITLE
create mp-b2b tests for merchant-addresses

### PR DESCRIPTION
**Pull request by m-alama**
### PR Description
create mp-b2b tests for merchant-addresses

### CODE REVIEW

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot [hunk 1]
```diff
@@ -0,0 +1,16 @@
+*** Settings ***
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot:1]: please rename the folder to 'merchant_endpoints'

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+Retrieves_merchant_addresses
+    When I send a GET request:  /merchants/${merchant_id}/merchant-addresses
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Each array element of array in response should contain property:    [data]    type
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:15]: Do we need to check the data for the current merchant with 'Each array element of array in response should contain property:'?
It seems that we have only 1 array element for every parameter, so we can use 'Response body parameter should be:' keyword

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    And Each array element of array in response should contain property:    [data]    id
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:16]: we can check the concrete id: "merchant_id":"MER000011"

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    countryName
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    address1
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    address2
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    address3
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    city
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    zipCode
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    latitude
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    longitude
+    And Each array element of array in response should contain nested property:    [data]    [links]    self
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:27]: It's better to use 'Response body has correct self link' keyword for link

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Retrieves_merchant_addresses
+    When I send a GET request:  /merchants/${merchant_id}/merchant-addresses
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Each array element of array in response should contain property:    [data]    type
+    And Each array element of array in response should contain property:    [data]    id
+    And Each array element of array in response should contain property:    [data]    attributes
+    And Each array element of array in response should contain property with value:    [data]    type    merchant-addresses
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    countryName
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:19]: we can check the country for this merchant: "default_country": "Germany"

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    And Each array element of array in response should contain nested property:    [data]    [attributes][addresses]    address1
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:20]: we can check that addresses parameters are not empty and if it's possible check their values, foe example:
Response body parameter should be:    [data][0][attributes][addresses][0][countryName]  ${default_country}

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot [hunk 7]
```diff
@@ -0,0 +1,27 @@
+*** Settings ***
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:7]: Please also add test with include: /merchants/${merchant_id}?include=merchant-addresses

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+
+*** Test Cases ***
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot:8]: Please add also negative test without merchant Id: /merchants//merchant-addresses
Need to check status code #400

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot (hunk 1 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot:1]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 3 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:16]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 4 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:27]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 5 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:19]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 6 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot:20]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot (hunk 8 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/negative.robot:8]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 7 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot]: Done

File: tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot (hunk 2 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/merchants_endpoints/merchant_addresses/positive.robot]: I'm just make it general as much as I can 
cuz for any reason if we change current merchant id 
tests will be success all time
but I did my best to test real values and make it general

File: tests/api/mp_b2b/glue/merchant_endpoints/merchant_addresses/positive.robot [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    And Response Should Contain The Array Larger Than a Certain Size:    [data][relationships]    0
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][merchant-addresses]
+    And Each Array Element Of Array In Response Should Contain Property:    [data][relationships][merchant-addresses][data]    type
+    And Each Array Element Of Array In Response Should Contain Property:    [data][relationships][merchant-addresses][data]    id
+    And Response body parameter should not be EMPTY:    [included]
+    And Each Array Element Of Array In Response Should Contain Property:    [included]    type
+    And Each Array Element Of Array In Response Should Contain Property:    [included]    id
+    And Each Array Element Of Array In Response Should Contain Property:    [included]    attributes
+    And Each Array Element Of Array In Response Should Contain Property:    [included]    links
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/merchant_endpoints/merchant_addresses/positive.robot:51]: Please add detail checks for included parameters, for example:
Response body parameter should be: [included][0][type] merchant-addresses

We need to check concrete data for the current merchant

**Review by vadym-uskov — APPROVED** [review]

**Pull request merged**